ext/ddtrace_profiling_native_extension/collectors_cpu_and_wall_time_worker.c

```diff
@@ -128,7 +133,7 @@ void collectors_cpu_and_wall_time_worker_init(VALUE profiling_module) {
   // https://bugs.ruby-lang.org/issues/18007 for a discussion around this.
   rb_define_alloc_func(collectors_cpu_and_wall_time_worker_class, _native_new);
 
-  rb_define_singleton_method(collectors_cpu_and_wall_time_worker_class, "_native_initialize", _native_initialize, 2);
+  rb_define_singleton_method(collectors_cpu_and_wall_time_worker_class, "_native_initialize", _native_initialize, 3);
```

Comment:
For context, the `2` to `3` change here is the number of positional arguments to the method. This is documented [here](https://github.com/ruby/ruby/blob/master/doc/extension.rdoc#method-and-singleton-method-definition-).

Effectively there’s three ways to “wire up” a C function to be called as a Ruby method:

* if you pass in a number >= 0 then that’s the number of positional arguments to the method. E.g. `def foo(a, b, c)` => `rb_define_method(..., foo_implementation, 3)` and `foo_implementation` is expected to have a signature that receives self + 3 other arguments, e.g. `VALUE foo_implementation(VALUE self, VALUE a, VALUE b, VALUE c)`.
* if you pass in `-1` then Ruby will pass the arguments as a C array
* if you pas in `-2` then Ruby will pass in the arguments as a Ruby array (e.g. it’s the C equivalent of `def foo(*args)`)

I moved it from `2` to `3` since I was passing the new `gc_profiling_enabled` value